app/Http/Controllers/BeatmapDiscussionPostsController.php

```diff
@@ -128,8 +128,7 @@ public function store()
             if ($resetNominations) {
                 $events[] = BeatmapsetEvent::NOMINATION_RESET;
             } else {
-                $disqualify = $discussion->beatmapset->isQualified() &&
-                    priv_check('BeatmapsetDisqualify', $discussion->beatmapset)->can();
+                $disqualify = priv_check('BeatmapsetDisqualify', $discussion->beatmapset)->can();
```

Comment:
although you added check in `disqualify` method, new disqualify event will always be generated because there's no check whether or not it's actually disqualified.